financial-connections/src/main/java/com/stripe/android/financialconnections/features/partnerauth/PartnerAuthViewModel.kt

```diff
@@ -168,41 +170,34 @@ internal class PartnerAuthViewModel @Inject constructor(
 
     fun onLaunchAuthClick() {
         setState { copy(authenticationStatus = Loading(value = Status(Action.AUTHENTICATING))) }
-        viewModelScope.launch {
-            awaitState().payload()?.authSession?.let {
-                postAuthSessionEvent(it.id, AuthSessionEvent.OAuthLaunched(Date()))
-                eventTracker.track(PrepaneClickContinue(PANE))
+
+        withState { state ->
+            val authSession = requireNotNull(state.payload()?.authSession) {
+                "Payload shouldn't be null when the user launches the auth flow"
```

Comment:
Any risk that this is ever null?

Reply:
it is very unlikely (since a session is created when initializing the pane)